free-3.md

```diff
@@ -1,49 +1,79 @@
+
+I really enjoy arranging music, and would really enjoy a tool that makes the process more efficient! 
+
+_How did you come up with it?_
+
+I came up with it one day when I was really frustrated with the painstaking process of translating musical thoughts into sheet music via MuseScore, and I did an angry keyboard smash. Then, I realized how cool it would be if I could tell Musecore the tempo of my song, and then press the computer keyboard keys that correspond to the notes I wanted to dictate. Ideally, the composition software would also compare how long I hold the key down (lets say 700 milliseconds) to the BPM (tempo) of the tempo I determined, to notate not only the notes themselves but the duration of the notes.
 
 ### Domain
-_Describe the project's domain in five words._
+Music Composition 
 
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+_How might the user interact with the language?_
+
+The language will provide a text input box that allows someone to type in the tempo in BPM, time signature, key, and clef. For example, the user could say that the song was in 96BPM per quarter note, in 4/4 time, in G Major, and in treble clef. Then, a new MuseScore file would be created, and prompt the user to decide how many voices/instruments were going to be involved in the piece. The user chooses one of the instruments, starts the metronome, and presses the keyboard keys representing the names of the notes he wants to notate, holding down the key presses for a long amount of time if the note was longer, and a short amount of time if the note was shorter. 
+
+_What does programming look like? Why is this the right way to interact with the problem domain?_ 
+
+I described a bit above about how programming would look like. I think this is the right way to interact with the problem domain, because I want the music composition process to be as fast and efficient as possible, even without a piano keyboard.
 
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
+_What might happen when a program runs? How does a program interact with the user?_
 
+The program prommpts the user for details regarding certain features of the music, including the clefs, key signature, time signature, tempo, etc. Then, as the user is pressing down the keys, the sheet music shows a faint image of the notes in the score. If the note was a half note (2 quarter notes) and the time signature was 4/4, the user would see the note appear as a quarter note at first, and then after he/she held it down too long to be considered a quarter note, it turns into a half note on the score, contuining to update the value of the note according to the pre-set time signature and BPM until the user releases that particular key.
+
```

Comment:
This is certainly very different from the DSLs we have encountered so far.  Naturally, the vocabulary seems like the computer keys themselves.  I guess the grammar would be the order that you type the keys, but also the duration of  each note.  It seems foreign to include aspects like "the amount of time you press a key" as part of the grammar, but I see nothing wrong with that idea.  It most certainly serves a specific domain, so I guess from our definition in class, you could definitely argue that this would be a DSL.